pandas/core/internals/blocks.py

```diff
@@ -1288,7 +1276,7 @@ def take_nd(self, indexer, axis, new_mgr_locs=None, fill_tuple=None):
         else:
             return self.make_block_same_class(new_values, new_mgr_locs)
 
-    def diff(self, n, axis=1):
+    def diff(self, n: int, axis: int = 1):
```

Comment:
I think can type the return as `-> "Block"` here

Reply:
Think this would still be nice as a follow up or another commit

Reply:
is using a string here the best practice?

Good to have you back

Reply:
Thanks! And yea you would want to use a string literal when the return type for a function is a type of the class itself

Mypy and related documentation will refer to this as a forward reference

Reply:
added